tests/unit/handler.unitTest.ts

```diff
@@ -164,6 +164,66 @@ describe("The lambda function handler", () => {
         expect(updateEuVehicleMock.mock.calls[0][1]).toEqual("m1");
       });
 
+      it("should call the /vehicles/add-provisional function with correct event payload", async () => {
+        const payload = {
+          msUserDetails: {
+            msUser: "dorel",
+            msOid: "12314234"
+          },
+          vin: "XMGDE02FS0H012345",
+          systemNumber: mockData[26].systemNumber,
+          techRecord: mockData[26].techRecord
+        };
+
+        const vehicleRecordEvent = {
+          path: "/vehicles/add-provisional/10000027",
+          pathParameters: {
+            systemNumber: "10000027"
+          },
+          resource: "/vehicles/add-provisional/{systemNumber}",
+          httpMethod: "POST",
+          body: JSON.stringify(payload)
+        };
+        const addProvisionalMock = jest.fn().mockImplementation(() => {
+          return Promise.resolve(new HTTPResponse(200, {}));
+        });
+        TechRecordsService.prototype.addProvisionalTechRecord = addProvisionalMock;
```

Comment:
Any reasons why we are mutating the method here?
We need to reset the mock after the test and I am not sure whether this is the correct way to mock this method.
I think we want to use jest.mock() automocking first to mock the service

Reply:
need your help on this.